core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java

```diff
@@ -457,4 +643,26 @@ private void workerLostHandler(long id) {
   private void registerNewWorkerConfHandler(long id, List<ConfigProperty> configList) {
     mWorkerConfigChecker.registerNewConf(id, configList);
   }
+
+  /**
+   * Sets the master id and hostname.
+   */
+  private void setMasterIdAndHostname() {
+    mMasterHostname = Configuration.get(PropertyKey.MASTER_HOSTNAME);
```

Comment:
With fault tolerant masters, will this property always be set to the machines hostname? I feel like this property may not always be the machine's hostname.

Reply:
The masterHostname here only be used for the leader master to identify a node and assign an id. I think if the value of MASTER_HOSTNAME is unchanged at runtime and no two masters have the same value, using MASTER_HOSTNAME as a representative of this node should be fine.

Reply:
Well, I don't think masters use `MASTER_HOSTNAME`. (this is a confusing part of the config) I think `MASTER_HOSTNAME` is only used for clients accessing the master to connect to the master.

I think this might be the first time the master uses the `MASTER_HOSTNAME` for its own hostname, so I don't think this will always be set correctly. Instead, maybe we should use: https://github.com/Alluxio/alluxio/blob/v1.7.1/core/common/src/main/java/alluxio/util/network/NetworkAddressUtils.java#L277

Reply:
In HA mode masters need their own `alluxio.master.hostname` so that they can register themselves with zookeeper